lib/Doctrine/ORM/EntityManager.php

```diff
@@ -349,12 +357,21 @@ public function detach($object)
      * @throws OptimisticLockException If a version check on an entity that
      *         makes use of optimistic locking fails.
      * @throws ORMException
+     * @throws LogicException
      */
     public function flush()
     {
+        if ($this->flushInProgress) {
+            throw new LogicException('EntityManager::flush must not be called inside another EntityManager::flush');
+        }
+
+        $this->flushInProgress = true;
+
         $this->errorIfClosed();
 
         $this->unitOfWork->commit();
+
+        $this->flushInProgress = false;
```

Comment:
This must go into a `finally` block otherwise it won't be reset in case of exception.

Reply:
Good catch. 👍  Fixed and covered by test.